### PR TITLE
Connection testing

### PR DESCRIPTION
**Pull request by jgraff2**
Corresponding changes: salesforce/refocus-collector-eval#10
refocus-collector-eval has been updated, this is ready to merge now.
There's a lot of changed files for this one. They are organized by commit.

### CODE REVIEW

**Comment by jgraff2** [comment]
Corresponding changes:
https://github.com/salesforce/refocus-collector-eval/pull/10

File: README.md [hunk 1]
```diff
@@ -113,13 +111,13 @@ Write your tests in the stubbed out `/transform/testTransform.js` file.
 
 ### toUrl
 
-If you need to do more complex transformations to generate a URL, rather than just simple variable substitutions, go to  /src/toUrl.js and implement your logic in the function which is already stubbed out in that file. The function must return a string. The function body has access to these variables:
+If you need to do more complex transformations to generate a URL, rather than just simple variable substitutions, go to  /connection/connection.js and implement your logic in the function which is already stubbed out in that file. The function must return a string. The function body has access to these variables:
```
Comment (iamigo) [inline README.md]: I think this section has to describe the process for defining the url whether it's using toUrl or basic substitutions.

File: examples/connection/basic/testConnection.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+   * Url (optional)
+   *
+   * Test your url by calling prepareUrl and check that the
+   * returned string is the expected url
+   */
+  describe('prepareUrl >', () => {
+   it('prepareUrl', () => {
+     const url = tu.prepareUrl();
+
+     expect(url).to.have.protocol('https');
```
Comment (iamigo) [inline examples/connection/basic/testConnection.js]: this is great!

File: examples/connection/concatenateAspects/connection.js [hunk 3]
```diff
@@ -1,25 +1,20 @@
...
- * DO NOT declare any functions OUTSIDE the body of the toUrl function
- *  itself.
- *
  */
 
 module.exports = {
 
   /**
-   * Creates an url string by concatenating the aspects names and including
+   * Creates a url string by concatenating the aspects names and including
```
Comment (iamigo) [inline examples/connection/concatenateAspects/connection.js]: thx!

File: runExample.sh [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 # Usage:
 # ./runExample -t transformExample
-# ./runExample -u toUrlExample
+# ./runExample -c connectionExample
```
Comment (iamigo) [inline runExample.sh]: I missed this getting into the project when you first added it. I agree that this script is very useful, but I'd prefer to see it written using node.js.

Not a deal-breaker, just a strong preference ;)

Comment (jgraff2) [reply runExample.sh]: Makes sense. It was originally just a consolidation of the shell commands I was already using to test the examples, and it was just for my own use. I'll create a story to convert it.

**Comment by jgraff2** [comment]
updated README.

**Pull request merged**